src/ios-deploy/ios-deploy.m

```diff
@@ -1436,6 +1438,59 @@ void remove_path(AMDeviceRef device) {
     assert(AFCConnectionClose(afc_conn_p) == 0);
 }
 
+void iter_dir(struct afc_connection *conn, char const *path, iter_callback callback) {
+    struct afc_directory *dir;
+    char *dirent;
+    
+    if(AFCDirectoryOpen(conn, path, &dir))
+        return;
+    
+    while(1) {
+        assert(AFCDirectoryRead(conn, dir, &dirent) == 0);
+        
+        if (!dirent)
+            break;
+        
+        if (strcmp(dirent, ".") == 0 || strcmp(dirent, "..") == 0)
+            continue;
+        
+        callback(conn, path, dirent);
+    }
+}
+
+void remove_path_callback(struct afc_connection *conn, char const *path, char *dirent) {
+    char subdir[255];
+    snprintf(subdir, 255, "%s/%s", path, dirent);
+    remove_path_recursively_conn(conn, subdir);
+}
+
+void remove_path_recursively_conn(struct afc_connection *conn, char const *path) {
+    int ret = AFCRemovePath(conn, path);
+    if(ret == 0 || ret == 8) {
+        // Successfully removed (it was empty) or does not exist
+        NSLogVerbose(@"Deleted %s", path);
+        return;
+    }
+    
+    iter_dir(conn, path, (iter_callback) remove_path_callback);
+    
+    assert(AFCRemovePath(conn, path) == 0);
```

Comment:
it make possible to make empty Documents folder without assert:

    iter_dir(conn, path, (iter_callback) remove_path_callback);
    ret = AFCRemovePath(conn, path);
    if (ret == 10)
    {
        // No permissions for deleting folder
        NSLogVerbose(@"No permissions for delete %s", path);
        return;
    }